ksqldb-rest-app/src/main/java/io/confluent/ksql/rest/server/computation/DistributingExecutor.java

```diff
@@ -241,6 +251,64 @@ public StatementExecutorResponse execute(
     }
   }
 
+  /**
+   * @return a list of warning messages for deprecated syntax statements
+   */
+  private List<KsqlWarning> getDeprecatedWarnings(
+      final MetaStore metaStore,
+      final ConfiguredStatement<?> statement
+  ) {
+    final List<KsqlWarning> deprecatedWarnings = new ArrayList<>();
+
+    if (isLeftOrOuterStreamStreamJoinWithoutGrace(metaStore, statement.getStatement())) {
+      deprecatedWarnings.add(new KsqlWarning(
+          "DEPRECATION NOTICE: Left/Outer stream-stream joins statements without a GRACE PERIOD "
+              + "will not be accepted in a future ksqlDB version.\n"
+              + "Please use the GRACE PERIOD clause as specified in "
+              + "https://docs.ksqldb.io/en/latest/developer-guide/ksqldb-reference/"
+              + "select-push-query/"
+      ));
+    }
+
+    return deprecatedWarnings;
+  }
+
+  private boolean isLeftOrOuterStreamStreamJoinWithoutGrace(
+      final MetaStore metaStore,
+      final Statement statement
+  ) {
+    // Check only CREATE_AS statements.
+    // Pull and push queries are not written in the command topic.
+    if (!(statement instanceof CreateAsSelect)) {
+      return false;
+    }
+
+    if (((CreateAsSelect) statement).getQuery().getFrom() instanceof Join) {
+      final Join join = (Join) ((CreateAsSelect) statement).getQuery().getFrom();
+
+      // Check join sources are streams only
```

Comment:
What about n-way joins: There could we one stream-stream join followed by a stream-table join.